test/kubeclient/kubeclient.go

```diff
@@ -123,6 +131,75 @@ func new(configBytes []byte, namespace string) (*KubeClient, error) {
 	}, nil
 }
 
+func (kc *KubeClient) CreateAzureCredentialsKubeSecret(ctx context.Context) error {
```

Comment:
It looks more complex than it should be. @squizzi already proposed several fixes and I tend to agree with him